### ISSUE TITLE
Global platform.txt is not supported

### ISSUE THREAD

**Issue by NicoHood** [arduino/arduino-cli#999]
### Describe the problem

Arduino IDE 1.x supports defining platform properties for all boards platforms via a "global `platform.txt`":

https://arduino.github.io/arduino-cli/dev/platform-specification/#global-platformtxt

Arduino CLI does not provide this feature.

Due to the deficiency in Arduino CLI, Arduino IDE 2.x does not provide this feature, but that deficiency is not documented in the Arduino Platform Specification.

### To reproduce

1. Select **File > Quit** (or **Arduino > Quit Arduino** for macOS users) from the Arduino IDE 2.x menus.
   All Arduino IDE windows will close.
1. Create a folder named `hardware` under the Arduino IDE 2.x installation folder.
   - **ⓘ** It is necessary to create this folder because, unlike Arduino IDE 1.x, the 2.x installation does not have such a folder.
   - **ⓘ** Arduino IDE is used for this demo because the information in the Arduino Platform Specification is specific to Arduino IDE and doesn't give any indication of how the feature might work with Arduino CLI if implemented.
1. Create a file named `platform.txt` in the `hardware` folder.
1. Add the following content to the `platform.txt` file:
   ```text
   recipe.hooks.sketch.prebuild.42.pattern=echo hello world
   recipe.hooks.sketch.prebuild.42.pattern.windows=cmd /c echo hello world
   ```
1. Start Arduino IDE 2.x.
1. Select **File > New Sketch** from the Arduino IDE menus.
1. Select **File > Preferences...** (or **Arduino IDE > Settings...** for macOS users) from the Arduino IDE menus.
   The "**Preferences**" dialog will open.
1. Check the box next to "**Show verbose output during: ☐ compile**" in the "**Preferences**" dialog.
1. Click the "**OK**" button.
   The "**Preferences**" dialog will close.
1. Select **Sketch > Verify/Compile** from the Arduino IDE menus.
1. Wait for the compilation to finish.
1. Examine the contents of the black "**Output**" panel at the bottom of the Arduino IDE window.

🐛 The output does not contain the "`hello world`" message that should have been produced by the build hook defined in the global `platform.txt` file.

### Expected behavior

All features described in the Arduino Platform Specification work.

When a feature is not available in some versions of Arduino development tools, this must be documented in the Arduino Platform Specification.

#### Original report

Not specified.

#### Last verified with

b9edb782a265b99878b4bce489c1751c4dcbee61

### Operating system

All

### Operating system version

Any

#### Additional reports

- https://github.com/arduino/arduino-cli/issues/999#issuecomment-278548207
- https://github.com/arduino/arduino-cli/issues/999#issuecomment-449580176
- https://github.com/arduino/arduino-cli/issues/999#issuecomment-1250019122
- https://forum.arduino.cc/t/global-platform-txt-on-mac-osx/1359912

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-cli/issues?q=)
- [X] I verified the problem still occurs when using the [nightly build](https://arduino.github.io/arduino-cli/dev/installation/#nightly-builds)
- [X] My report contains all necessary details

**Comment by per1234** [comment]
If I understand the request correctly, we now have this feature, but only for Arduino IDE:
https://arduino.github.io/arduino-cli/latest/platform-specification/#global-platformtxt

**Comment by ubidefeo** [comment]
@per1234 
you beat me to the punch as usual :D

**Comment by fabianoriccardi** [comment]
I agree with the previous posts, I would be great to have this feature back or, at least, another system to run pre-/ post- scripts before/after a build.

Is still planned?

**Comment by per1234** [comment]
Thanks for your request @NicoHood. This was actually resolved long ago by https://github.com/arduino/arduino-cli/pull/472. However, that was not known due to a misimpression that the feature was specific to the IDE-bundled packages folder unique to the Arduino IDE 1.x application. It turns out that global `platform.txt` files are (and always were) recognized in any of the packages folders; not the Arduino IDE 1.x bundled packages folder exclusively. Thus, we now understand that this issue is resolved.

The misinformation in the Arduino Platform Specification regarding the scope of the global `platform.txt` file has been corrected: https://github.com/arduino/arduino-cli/pull/2952. In addition, the locations of all the packages folders have been documented:

https://arduino.github.io/arduino-cli/dev/platform-specification/#global-platformtxt
